src/librustc_typeck/diagnostics.rs

```diff
@@ -733,6 +748,38 @@ fn some_func(x: &mut i32) {
 ```
 "##,
 
+E0071: r##"
+You tried to use a structure initialization with a non-structure type.
+Example of erroneous code:
+
+```
+enum Foo { f };
+
+let u = Foo::f { value: 0i32 }; // error: Foo:f isn't a structure!
+// or even simpler:
+let u = t { random_field: 0i32 }; //error: t isn't a structure!
```

Comment:
It's not been defined .. oh, I see

(make that explicit "or even simpler, if the structure wasn't defined at all")